WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalConnectionFailedUpdateAddress.swift

```diff
@@ -51,6 +58,13 @@ final class CardPresentModalConnectingFailedUpdateAddress: CardPresentPaymentsMo
             return retrySearchAction()
         }
         openWCSettingsAction()
+        if let adminURL = wcSettingsAdminURL {
+            wcSettingsWebViewModel = .init(webViewURL: adminURL, onCompletion: { [weak self] in
+                guard let self else { return }
+                wcSettingsWebViewModel = nil
+                retrySearchAction()
+            })
+        }
```

Comment:
I think this should be done in a specialised handler implementation, since it's not used (AIUI) in the UIKit implementation...

Done in 56ab565

Reply:
Great point! Thanks for the update 👍